DB/pgsql.php

```diff
@@ -1110,6 +1127,29 @@ function _checkManip($query)
                 || parent::_checkManip($query));
     }
 
+    // }}}
+    // {{{ _resultId()
+
+    /**
+     * Returns a numeric value identifying the result, used for maintaining
+     * indexes. Required by migration from resources to 'PgSql\Result'
+     * types with PHP 8.1.
+     *
+     * @param mixed $result  the query result resource or PgSql\Result
+     *
+     * @return int
+     *
+     * @access protected
+     */
+    function _resultId($result)
+    {
+        return
+            is_resource($result)
+                ? ( function_exists('get_resource_id')
+                    ? get_resource_id($result)
+                    : (int)$result
+                ) : spl_object_id($result);
```

Comment:
Thanks @DanCld for your changes.
But for backwards compatibility, there is still a little changed needed.
Both functions `spl_object_id()` and `get_resource_id()` are not compatible with PHP5.
Could you check first, if the function `spl_object_id()` exists, and if not returning the integer casted `$result` instead, as you did it with the `get_resource_id()` function, please?

Then i´m going to merge your request into the main branch.

Reply:
I don't think that's needed. The decision tree for PHP5 is:

1. is_resource(...) will return true for PHP < 8.1
2. get_resource_id() is defined for PHP >= 8.0, so the _false_ path is being picked;
3. consequently, (int)$result is evaluated

For PHP == 8.0, is_resource(...) will return true and get_resource_id() is available, so this path is chosen.

For PHP >= 8.1, is_resource(...) will return false, while spl_object_id() is defined starting with PHP7, so no worries here.

Reply:
@DanCld Sorry, you're right. I misunderstood that pgsql returns a resource before PHP8.

I agree and going to merge your PR later, when I'm at home.

Thank you for your great contribution.